### PR TITLE
Add circuit breaker to http-result generation

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

**Comment by mfussenegger** [comment]
I was also thinking about integrating http://openjdk.java.net/projects/code-tools/jol/ - so that we'd have more accurate instance sizes. But it's GPL licensed :(

File: sql/src/main/java/io/crate/breaker/CollectionSizeEstimator.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        this.innerEstimator = innerEstimator;
+    }
+
+    @Override
+    public long estimateSize(@Nullable Object value) {
+        if (value == null) {
+            return 8;
+        }
+        if (value.getClass().isArray()) {
+            return sizeFromArray((Object[]) value);
```
Comment (seut) [inline sql/src/main/java/io/crate/breaker/CollectionSizeEstimator.java]: why can this happen if a CollectionSizeEstimator is only used for `type instanceof CollectionType`?

File: sql/src/main/java/io/crate/executor/transport/task/ShowCreateTableTask.java [hunk 2]
```diff
@@ -47,7 +48,7 @@ public void execute(BatchConsumer consumer, Row parameters) {
         Row1 row;
         try {
             CreateTable createTable = MetaDataToASTNodeResolver.resolveCreateTable(tableInfo);
-            row = new Row1(SqlFormatter.formatSql(createTable));
+            row = new Row1(new BytesRef(SqlFormatter.formatSql(createTable)));
```
Comment (seut) [inline sql/src/main/java/io/crate/executor/transport/task/ShowCreateTableTask.java]: is this related to that commit? if not maybe put it into a different commit?

File: sql/src/test/java/io/crate/breaker/RowAccountingTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    public void testCircuitBreakingWorks() throws Exception {
+        RowAccounting rowAccounting = new RowAccounting(Collections.singletonList(DataTypes.INTEGER),
+            new RamAccountingContext(
+                "test",
+                new MemoryCircuitBreaker(
+                    new ByteSizeValue(10, ByteSizeUnit.BYTES), 1.01, Loggers.getLogger(RowAccountingTest.class))
+            ));
+
+        expectedException.expect(CircuitBreakingException.class);
+        // internal flush buffer is 2MB, so it doesn't immediately trip after the first row
```
Comment (seut) [inline sql/src/test/java/io/crate/breaker/RowAccountingTest.java]: you could change ( and reset afterwards) the flush buffer size... not sure if its worth, should be pretty fast already or?

File: sql/src/main/java/io/crate/executor/transport/task/ShowCreateTableTask.java (hunk 2 above)
Comment (mfussenegger) [inline sql/src/main/java/io/crate/executor/transport/task/ShowCreateTableTask.java]: It's kinda related: Now the SizeEstimator for the string type is used which expects BytesRef instead of String.
